app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/qhybrid/QHybridSupport.java

```diff
@@ -0,0 +1,821 @@
+        addSupportedService(UUID.fromString("0000180a-0000-1000-8000-00805f9b34fb"));
+        addSupportedService(UUID.fromString("00001800-0000-1000-8000-00805f9b34fb"));
+        addSupportedService(UUID.fromString("0000180f-0000-1000-8000-00805f9b34fb"));
+        IntentFilter commandFilter = new IntentFilter(QHYBRID_COMMAND_CONTROL);
+        commandFilter.addAction(QHYBRID_COMMAND_UNCONTROL);
+        commandFilter.addAction(QHYBRID_COMMAND_SET);
+        commandFilter.addAction(QHYBRID_COMMAND_VIBRATE);
+        commandFilter.addAction(QHYBRID_COMMAND_UPDATE);
+        commandFilter.addAction(QHYBRID_COMMAND_NOTIFICATION);
+        commandFilter.addAction(QHYBRID_COMMAND_UPDATE_SETTINGS);
+        commandFilter.addAction(QHYBRID_COMMAND_OVERWRITE_BUTTONS);
+        LocalBroadcastManager.getInstance(getContext()).registerReceiver(commandReceiver, commandFilter);
+        fillResponseList();
+
+        helper = new PackageConfigHelper(GBApplication.getContext());
+
+        IntentFilter globalFilter = new IntentFilter();
+        globalFilter.addAction(QHYBRID_ACTION_SET_ACTIVITY_HAND);
+        GBApplication.getContext().registerReceiver(globalCommandReceiver, globalFilter);
+    }
+
+    private boolean supportsActivityHand() {
+        switch (modelNumber) {
+            case "HL.0.0":
+                return false;
+            case "HW.0.0":
+                return true;
+        }
+        throw new UnsupportedOperationException();
+    }
+
+    private boolean supportsExtendedVibration() {
+        switch (modelNumber) {
+            case "HL.0.0":
+                return false;
+            case "HW.0.0":
+                return true;
+        }
+        throw new UnsupportedOperationException();
+    }
+
+    private void fillResponseList() {
+        Class<? extends Request>[] classes = new Class[]{
+                BatteryLevelRequest.class,
+                GetStepGoalRequest.class,
+                GetVibrationStrengthRequest.class,
+                GetCurrentStepCountRequest.class,
+                OTAEnterRequest.class,
+                GoalTrackingGetRequest.class,
+                ActivityPointGetRequest.class,
+                GetCountdownSettingsRequest.class
+        };
+        for (Class<? extends Request> c : classes) {
+            try {
+                c.getSuperclass().getDeclaredMethod("handleResponse", BluetoothGattCharacteristic.class);
+                Request object = c.newInstance();
+                byte[] sequence = object.getStartSequence();
+                if (sequence.length > 1) {
+                    responseFilters.put((int) object.getStartSequence()[1], object);
+                    Log.d("Service", "response filter " + object.getStartSequence()[1] + ": " + c.getSimpleName());
+                }
+            } catch (NoSuchMethodException | IllegalAccessException | InstantiationException e) {
+                Log.d("Service", "skipping class " + c.getName());
```

Comment:
Please make these LOG.warn() instead, having such things in logfiles is really helpful at times.